src/components/MeetupGallery.astro

```diff
@@ -0,0 +1,193 @@
+---
+const props = Astro.props
+---
+
+
+<div class="main-container">
+      <div class="container">
+        {
+          props.currentAlbum && props.currentAlbum.length > 0 &&
+          props.currentAlbum
+          .slice(0, 3)
+          .map((photo,index)  => (
+              <div class={`photo-container photo-${index+1}`}>
+                <img
+                  src={`${props.source}/${photo}`}
+                //   class=""
+                  loading="lazy"
+                />
+              </div>
+            ))
+        }
+      </div>
+  </div>
+
+  <script define:vars={{ props }}>
+
+    const imageArray = getSplitArray(props.currentAlbum, 3);
+    let counters = [0, 0, 0];
+    let isMouseOver = false;
+
+
+    startAnimation();
+    addEventListenerOnImages();
+
+    function startAnimation(){
+      setInterval(() => {
+        if(!isMouseOver){
+          changeImages();
+        }
+      }, 7000);
```

Comment:
move the duration to a variable, or even better,a prop that can be passed to the component from astro. if no prop is passed set a default value. probably 3000ms